src/tracing/mod.rs

```diff
@@ -229,12 +230,80 @@ impl<P: EthereumProvider + Send + Sync + Clone> Tracer<P> {
         Err(EthApiError::TransactionNotFound(transaction_hash))
     }
 
+    /// Debugs a transaction request by tracing it using the provided tracing options.
+    ///
+    /// This function returns an error if the tracing options are not supported or if there is an issue
+    /// with the EVM environment or transaction execution.
+    pub fn debug_transaction_request(self, request: &TransactionRequest) -> TracerResult<GethTrace> {
+        // Attempt to get Geth tracing options from the provided tracing options.
```

Comment:
I'd like us to consider a little more if we can't combine this function in some way with `trace_geth` in order to remove the boiler plate code.

Reply:
mhhhh I wanted too but I feel that the logic is very different and so I'm afraid it would add confusion in general to mix up the functions... even if I agree that we could find some repetitions in there